### PR TITLE
Use `@cspotcode/source-map-support` in `@babel/register`

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Now we no longer need this test since `@babel/core` no longer depends on it.

### CODE REVIEW

File: packages/babel-register/test/browserify.js [hunk 1]
```diff
@@ -1,25 +0,0 @@
-import browserify from "browserify";
-import path from "path";
-import vm from "vm";
-import { fileURLToPath } from "url";
-
-describe("browserify", function () {
-  it("@babel/register may be used without breaking browserify", function () {
```
Comment (nicolo-ribaudo) [inline packages/babel-register/test/browserify.js:7]: Can you elaborate more on why this test is not needed anymore?

Comment (liuxingbaoyu) [reply packages/babel-register/test/browserify.js:7]: The reason why this test was added is that `@babel/core` previously depended on `@babel/register`, and we hope that `@babel/core` can be used in the browser through `browserify`.
Now that it is no longer relied on, `@babel/register` itself does not need to be ensured to run in the browser.

Comment (nicolo-ribaudo) [reply packages/babel-register/test/browserify.js:7]: Ok thanks! Could we keep it for Babel 7?

Comment (liuxingbaoyu) [reply packages/babel-register/test/browserify.js:7]: OK, I'll leave it as a TODO.

Comment (nicolo-ribaudo) [reply packages/babel-register/test/browserify.js:7]: Can also just use `itBabel7`

**Review by nicolo-ribaudo — APPROVED** [review]

**Review by JLHwung — APPROVED** [review]

**Pull request merged**